analytics_data_api/v0/views/courses.py

```diff
@@ -30,11 +30,17 @@ def get(self, request, *args, **kwargs):
         timezone = utc
 
         if start_date:
-            start_date = datetime.datetime.strptime(start_date, settings.DATE_FORMAT)
+            try:
```

Comment:
Nit: I guess both the `state_date` and `end_date` parsing could be refactored into a single method.